**To get details of the dataset**

In [15]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

df= pd.read_csv('/kaggle/input/movie-title/titles.csv')
df.head( )
df.info()
df.describe()



<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5850 entries, 0 to 5849
Data columns (total 15 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   id                    5850 non-null   object 
 1   title                 5849 non-null   object 
 2   type                  5850 non-null   object 
 3   description           5832 non-null   object 
 4   release_year          5850 non-null   int64  
 5   age_certification     3231 non-null   object 
 6   runtime               5850 non-null   int64  
 7   genres                5850 non-null   object 
 8   production_countries  5850 non-null   object 
 9   seasons               2106 non-null   float64
 10  imdb_id               5447 non-null   object 
 11  imdb_score            5368 non-null   float64
 12  imdb_votes            5352 non-null   float64
 13  tmdb_popularity       5759 non-null   float64
 14  tmdb_score            5539 non-null   float64
dtypes: float64(5), int64(

release_year      runtime      seasons   imdb_score    imdb_votes  \
count   5850.000000  5850.000000  2106.000000  5368.000000  5.352000e+03   
mean    2016.417094    76.888889     2.162868     6.510861  2.343938e+04   
std        6.937726    39.002509     2.689041     1.163826  9.582047e+04   
min     1945.000000     0.000000     1.000000     1.500000  5.000000e+00   
25%     2016.000000    44.000000     1.000000     5.800000  5.167500e+02   
50%     2018.000000    83.000000     1.000000     6.600000  2.233500e+03   
75%     2020.000000   104.000000     2.000000     7.300000  9.494000e+03   
max     2022.000000   240.000000    42.000000     9.600000  2.294231e+06   

       tmdb_popularity   tmdb_score  
count      5759.000000  5539.000000  
mean         22.637925     6.829175  
std          81.680263     1.170391  
min           0.009442     0.500000  
25%           2.728500     6.100000  
50%           6.821000     6.900000  
75%          16.590000     7.537500  
max        2274.044000    10.000000

**Cleaning the Data**

In [16]:
df.isna().sum()


id                         0
title                      1
type                       0
description               18
release_year               0
age_certification       2619
runtime                    0
genres                     0
production_countries       0
seasons                 3744
imdb_id                  403
imdb_score               482
imdb_votes               498
tmdb_popularity           91
tmdb_score               311
dtype: int64

No Misiing or NULL value found .

In [23]:
df['genres'].head()

5                                ['comedy', 'european']
26                                           ['comedy']
35                                           ['comedy']
44                ['scifi', 'action', 'crime', 'drama']
45    ['animation', 'family', 'comedy', 'fantasy', '...
Name: genres, dtype: object

**checking for duplicated data**

In [25]:
df.duplicated().sum()

0

In [26]:
df.describe()

release_year      runtime      seasons   imdb_score    imdb_votes  \
count   1667.000000  1667.000000  1667.000000  1667.000000  1.667000e+03   
mean    2017.355729    38.767846     2.337732     6.991482  1.970630e+04   
std        5.281628    16.944673     2.922864     1.067871  7.666425e+04   
min     1969.000000     0.000000     1.000000     2.000000  5.000000e+00   
25%     2016.000000    25.000000     1.000000     6.400000  6.060000e+02   
50%     2019.000000    41.000000     1.000000     7.100000  2.270000e+03   
75%     2020.000000    49.000000     3.000000     7.700000  8.862000e+03   
max     2022.000000   178.000000    42.000000     9.500000  1.775990e+06   

       tmdb_popularity   tmdb_score  
count      1667.000000  1667.000000  
mean         31.121982     7.464122  
std          96.320717     1.049718  
min           0.600000     1.000000  
25%           4.375500     7.000000  
50%           9.850000     7.500000  
75%          24.539500     8.100000  
max        2226.231000    10.000000

**Summary**

1. We have a dataframe consisting of total 5850 rows and 15 columns.
2. Our datset looks little bit tidy with no Nans and nor duplicated values.
3. Description , age certification , production countries and seasons won't be so useful in our analysis so we will be dropping them.
4. imdb votes can be better categorized for proper analysis.
5. Genres are having extra space after comma that need to be removed to make it look clean. 
   

In [27]:
df.head()

id                             title  type  \
5   ts22164      Monty Python's Flying Circus  SHOW   
26  ts45948  Monty Python's Fliegender Zirkus  SHOW   
35  ts20681                          Seinfeld  SHOW   
44  ts22082                      Knight Rider  SHOW   
45  ts21715                  Thomas & Friends  SHOW   

                                          description  release_year  \
5   A British sketch comedy series with the shows ...          1969   
26  Monty Python's Fliegender Zirkus consisted of ...          1972   
35  A stand-up comedian and his three offbeat frie...          1989   
44  Michael Long, an undercover police officer, is...          1982   
45  Thomas & Friends is a British children's telev...          1984   

   age_certification  runtime  \
5              TV-14       30   
26             TV-MA       43   
35             TV-PG       24   
44             TV-PG       51   
45              TV-Y       10   

                                               genres production_countries  \
5                              ['comedy', 'european']               ['GB']   
26                                         ['comedy']                   []   
35                                         ['comedy']               ['US']   
44              ['scifi', 'action', 'crime', 'drama']               ['US']   
45  ['animation', 'family', 'comedy', 'fantasy', '...               ['GB']   

    seasons    imdb_id  imdb_score  imdb_votes  tmdb_popularity  tmdb_score  
5       4.0  tt0063929         8.8     73424.0           17.617       8.306  
26      1.0  tt0202477         8.1      2151.0            1.487       7.000  
35      9.0  tt0098904         8.9    308824.0          130.213       8.301  
44      4.0  tt0083437         6.9     34115.0           50.267       7.500  
45     24.0  tt0086815         6.5      5104.0           42.196       6.500

--------    **DATA CLEANING INITIATED**    --------


*Dropping "Description" , " age certification" ,"production countries" ,"seasons"*

In [28]:
#making list of Colmn to be dropped
colm = ['description','age_certification','production_countries','seasons']

# dropping columns and confirming changes
df.drop(colm, axis = 1, inplace = True)
df.columns

Index(['id', 'title', 'type', 'release_year', 'runtime', 'genres', 'imdb_id',
       'imdb_score', 'imdb_votes', 'tmdb_popularity', 'tmdb_score'],
      dtype='object')

In [29]:
df.head()

id                             title  type  release_year  runtime  \
5   ts22164      Monty Python's Flying Circus  SHOW          1969       30   
26  ts45948  Monty Python's Fliegender Zirkus  SHOW          1972       43   
35  ts20681                          Seinfeld  SHOW          1989       24   
44  ts22082                      Knight Rider  SHOW          1982       51   
45  ts21715                  Thomas & Friends  SHOW          1984       10   

                                               genres    imdb_id  imdb_score  \
5                              ['comedy', 'european']  tt0063929         8.8   
26                                         ['comedy']  tt0202477         8.1   
35                                         ['comedy']  tt0098904         8.9   
44              ['scifi', 'action', 'crime', 'drama']  tt0083437         6.9   
45  ['animation', 'family', 'comedy', 'fantasy', '...  tt0086815         6.5   

    imdb_votes  tmdb_popularity  tmdb_score  
5      73424.0           17.617       8.306  
26      2151.0            1.487       7.000  
35    308824.0          130.213       8.301  
44     34115.0           50.267       7.500  
45      5104.0           42.196       6.500

**categorizing imdb_score column**

We would cut the "imdb_votes" values and make 4 categories: popular average
below_avg not_popular to describe it more using catigorize_col() function
provided above.


In [ ]:
coding for popular , not popular will be done here

**we'd split genres into a list and then
explode our dataframe to have only one
genre per row for ezch movie**

In [ ]:
# split the strings into lists
df['genres'] = df['genres'].str.split(', ')
# explode the lists
df = df.explode('genres').reset_index(drop=True)
df.head()
